packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -23,18 +25,32 @@ type DropdownMenuContextValue = {
   onOpenToggle(): void;
 };
 
+const SubmenuContext = React.createContext<boolean | undefined>(undefined);
 const [DropdownMenuProvider, useDropdownMenuContext] = createContext<DropdownMenuContextValue>(
   DROPDOWN_MENU_NAME
 );
 
-type DropdownMenuOwnProps = {
+type DropdownMenuOwnProps = React.ComponentProps<typeof DropdownMenuImpl>;
+
+const DropdownMenu: React.FC<DropdownMenuOwnProps> = (props) => {
+  const parentSubmenuContext = React.useContext(SubmenuContext);
+  return (
+    <SubmenuContext.Provider value={parentSubmenuContext === undefined ? false : true}>
```

Comment:
Also.... 💭

```jsx
<SubmenuContext.Provider value={Boolean(parentMenuContext)}>
```

I remember we did this `undefined` check for `ContextMenu` because we needed to check if it was `undefined` to determine `isRoot` but we don't need that in `DropdownMenu` right?

Reply:
The same applies here, we need `undefined` to determine the root, else it'll always fall one way or the other.